drivers/zfs/zfs.go

```diff
@@ -406,10 +406,20 @@ func (d *Driver) Get(id string, options graphdriver.MountOpts) (_ string, retErr
 		return "", errors.Wrap(err, "error creating zfs mount")
 	}
 
-	// this could be our first mount after creation of the filesystem, and the root dir may still have root
-	// permissions instead of the remapped root uid:gid (if user namespaces are enabled):
-	if err := os.Chown(mountpoint, rootUID, rootGID); err != nil {
-		return "", fmt.Errorf("error modifying zfs mountpoint (%s) directory ownership: %v", mountpoint, err)
+	readWrite := true
+	for _, option := range options.Options {
+		if option == "ro" {
+			readWrite = false
+			break
+		}
+	}
+
+	if readWrite {
```

Comment:
wouldn't this introduce a problem when namespaces are used since we skip the chown now?

I think what we need to do is:

- Mount filtering the `ro` flag
- Do the chown
- If `ro` was initially specified then remount `ro` (using `MS_REMOUNT`)

Reply:
Changed as requested. Please review.